test/webaudio/audioworklet_post_function.c

```diff
@@ -6,28 +6,121 @@
 // and the Audio Worklet thread using the
 // emscripten_audio_worklet_post_function_*() API.
 
-// This event will fire on the main thread.
-void MessageReceivedOnMainThread(int d, int e, int f) {
-  printf("MessageReceivedOnMainThread: d=%d, e=%d, f=%d\n", d, e, f);
-  assert(!emscripten_current_thread_is_audio_worklet());
-  assert(d == 1 && e == 2 && f == 3);
-#ifdef REPORT_RESULT
-  REPORT_RESULT(1); // test succeeded, were able to post a message from main thread to audio thread and back!
-#endif
+// This test scenario consists of two steps
+//   1. post function from main thread to audio worklet thread, with testingAudioWorklet = true
+//   2. post function from audio worklet thread to main thread, with testingAudioWorklet = false
+volatile bool testingAudioWorklet = true;
+volatile int success = 0;
```

Comment:
Use `_Atomic` rather than `volatile`?   Since that is better defined for multi-threaded access I believe.

Reply:
I agree with you